profiles/killbill/src/main/java/org/killbill/billing/server/modules/KillbillServerModule.java

```diff
@@ -106,6 +116,15 @@ protected void configure() {
         configurePushNotification();
 
         bind(new TypeLiteral<OSGIServiceRegistration<Healthcheck>>() {}).to(DefaultHealthcheckPluginRegistry.class).asEagerSingleton();
+
+        // Needed because changes in KillbillHealthcheck from Guice @Inject(optional=true) to javax @Inject + @Nullable
```

Comment:
Subtle! 😅 

We should do some additional manual testing to verify all of this plumbing still works as expected. For instance, service discovery can be tested using the Eureka bundle (https://github.com/killbill/killbill-platform/tree/work-for-release-0.23.x/osgi-bundles/bundles/eureka). @vnandwana can help if needed (he had to do similar testing when doing the metrics work).

Reply:
Noted.

Reply:
@pierre tracked [here](https://github.com/killbill/killbill/issues/1764), although I'm not sure what do you mean by ".... service discovery can be tested using the Eureka bundle".